services/consensuscontext/fetch_transactions.go

```diff
@@ -0,0 +1,29 @@
+package consensuscontext
+
+import (
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"time"
+	"fmt"
+)
+
+func (s *service) fetchTransactions(input *services.GetTransactionsForOrderingInput, minimumTransactionsInBlock int, belowMinimalBlockDelayMillis uint32) (*services.GetTransactionsForOrderingOutput, error) {
+
+	fmt.Println("Fetch transactions")
+	var proposedTransactions *services.GetTransactionsForOrderingOutput = nil
+	proposedTransactions, err := s.transactionPool.GetTransactionsForOrdering(input)
+	if err != nil {
+		return nil, err
+	}
+	txCount := len(proposedTransactions.SignedTransactions)
+	if txCount >= minimumTransactionsInBlock {
+		return proposedTransactions, nil
+	}
+	// TODO: How to test that Sleep() was called
+	time.Sleep(time.Duration(belowMinimalBlockDelayMillis) * time.Millisecond)
```

Comment:
if we need to run something twice with a delay maybe we should use a probe (see slack / block storage Get..Header functions), then look at tests on how we make sure the probe was used

Reply:
Note for me: 
blockstorage/service.go:104 - GetTransactionBlockHeader()
get_block_headers.go:55 - timeoutError

Reply:
Keeping Sleep() for now until we decide on the correct mechanism for syncing between services (e.g. context)